mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -72,6 +72,30 @@ where
             bucket_name: bucket_name.to_owned(),
         }
     }
+
+    pub async fn verify_cache_valid(&self) -> ObjectClientResult<(), PutObjectError, Client::ClientError> {
+        let object_key = format!("{}/_mountpoint_cache_metadata", &self.prefix);
+        // This data is human-readable, and not expected to be read by Mountpoint.
+        // The file format used here is NOT stable.
+        // For now, let's just include the data that's guaranteed to be correct as it's what
+        // calculates the prefix.
+        let data = format!(
+            "source_bucket={}\nblock_size={}",
```

Comment:
We don't actually put a restriction that the given directory bucket must be used to cache data for a single source bucket. Nothing critical will happen, if customer tries to do that (we'll just overwrite the object here), but it may get misleading (this object tells that a bucket is used to cache data for bucket A, while it also holds bucket B's data ).

Maybe we can try creating an empty object instead? Would it be sufficient as a check?